### PR TITLE
Initial implementation for multiple outputs

### PR DESCRIPTION
**Pull request by jipperinbham**
This is sort of a phase one to create an `outputs` configuration in order to send metrics to more than InfluxDB and begins to address https://github.com/influxdb/telegraf/issues/38.

I'd love some feedback here!

### CODE REVIEW

File: cmd/telegraf/telegraf.go [hunk 1]
```diff
@@ -8,8 +8,9 @@ import (
 	"os/signal"
 	"strings"
 
-	"github.com/influxdb/telegraf"
 	_ "github.com/influxdb/telegraf/plugins/all"
+	"github.com/jipperinbham/telegraf"
```
Comment (sparrc) [inline cmd/telegraf/telegraf.go]: These will need to be fixed before merging this PR

File: cmd/telegraf/telegraf.go [hunk 2]
```diff
@@ -8,8 +8,9 @@ import (
...
+	_ "github.com/jipperinbham/telegraf/outputs/all"
```
Comment (sparrc) [inline cmd/telegraf/telegraf.go]: change back to influxdb

**Comment by sparrc** [comment]
@jipperinbham Hi JP, if you want to fix the 2 comments I made and the branch conficts, then I can test this out and merge it!

**Comment by jipperinbham** [comment]
@sparrc I've updated the import paths to point to `influxdb` and rebased from master. Please let me know if something isn't building or running properly.

**Comment by sparrc** [comment]
@jipperinbham It looks like there may have been a merge problem, see ci failure above:

```./config.go:220: undefined: ErrInvalidConfig Action failed: go build```

**Comment by jipperinbham** [comment]
sorry about that @sparrc, should be good to test now

**Comment by sparrc** [comment]
@jipperinbham no prob!

This looks good to me, I like the design and fits with telegraf well.

Two questions, and maybe I'm just not 100% understanding the code, but should the sample and test TOML files be changed from [influxdb] to [outputs] ?

2nd: is there any validation that a valid output was specified? not that it needs to validate that the output destination is connectable, but maybe some validation that the configured output exists as an "out plugin" in telegraf?

PS: you can find those files in `etc/config.sample.toml` and `testdata/influx.toml`

**Comment by jipperinbham** [comment]
re: the sample TOML files, I held off on updating those in case there were some changes I needed to make but will do that now

re: output validation, I'm pretty sure the following covers it

https://github.com/jipperinbham/telegraf/blob/outputs-phase1/agent.go#L83-L86

**Comment by sparrc** [comment]
@jipperinbham I think you've accidentally added the `telegraf` binary to your commit, once that's removed I think the PR is good to go :+1:

**Comment by sparrc** [comment]
@jipperinbham I'm anxious to get this merged, are you done with this PR? I think last we left it, you needed to update the Connect() function?

**Comment by jipperinbham** [comment]
@sparrc yes, it should be good to go. Sorry I forgot to /cc you on that last commit. 

I couldn't fully test because I hit the line protocol change for deprecating ints but all the config and tags looked good.

**Comment by sparrc** [comment]
@jipperinbham I tried to merge this in but had some major conflicts, so I had to revert it, unfortunately I don't think that I'm able to re-open this PR.

Sorry for the hassle, but could you please rebase your changes and re-open a new PR?

**Comment by jipperinbham** [comment]
@sparrc no problem, I'll rebase and get things cleaned up and submit a new PR

**Comment by sparrc** [comment]
thanks @jipperinbham! Just to warn you, that is probably going to be a pretty hairy merge/rebase, there have been some pretty large changes in agent.go and config.go.

Let me know if you need a hand and we can walk through it together on Google hangouts if need be

**Comment by jipperinbham** [comment]
It looks like the issue is here, https://github.com/influxdb/telegraf/commit/29e8ce68e44daf4b5e5af6c9deb533bddcab912a, which I'm not entirely sure how to handle since we moved decided to move the Tags to the InfluxDB output and out of Config.

Not entirely sure how to proceed with the changes needed...

**Comment by sparrc** [comment]
@jipperinbham yes, this is what I was afraid of, I think I may have been wrong to suggest moving that into InfluxDB, and that the Tags should in fact stay in Config.

**Comment by jipperinbham** [comment]
no worries, I can easily look back through the git commits to move it back, it also makes cleaning up these merge conflicts possible without touching what others have added.

**Pull request merged**